library/mono

```diff
@@ -23,4 +21,13 @@ onbuild: git://github.com/mono/docker@adc7a3ec47f7d590f75a4dec0203a2103daf8db0 3
 3.8.0-onbuild: git://github.com/mono/docker@adc7a3ec47f7d590f75a4dec0203a2103daf8db0 3.8.0/onbuild
 3.8-onbuild: git://github.com/mono/docker@adc7a3ec47f7d590f75a4dec0203a2103daf8db0 3.8.0/onbuild
 
-alpha: git://github.com/mono/docker@f639c9ba71c3e3601c079b353e12351bd008b11e alpha
+4.0.1: git://github.com/mono/docker@0d3556995aa47043059d60c42321e8ccaf173363 4.0.0
+4.0.0: git://github.com/mono/docker@0d3556995aa47043059d60c42321e8ccaf173363 4.0.0
```

Comment:
`4.0.0` _and_ `4.0.1`?

Also, https://github.com/mono/docker/blob/0d3556995aa47043059d60c42321e8ccaf173363/4.0.0/Dockerfile#L13-L15 seems to imply that it's a snapshot of the `4.0.0` release, but we're getting `4.0.1` out of it?  Are `3.8.0` and `3.12.0` doing the same thing?

https://github.com/docker-library/official-images#instruction-format is probably the closest thing we have to relevant docs here, specifically:

> For example, if a Git tag is changed on the referenced repository to point to another commit, **the image will not be rebuilt**. Instead, either create a new tag (or reference an exact commit) and submit a pull request.

ie, the only ways our system will rebuild this image are via a pull request to this repository updating the commit reference or a rebuild of the base image (requiring a dependency rebuild), so we prefer to see builds either be repeatable, or fail (so APT pinning here would be a preferable way to handle this, like we do in a number of other repos)